plugins/inputs/proxysql/proxysql.go

```diff
@@ -0,0 +1,354 @@
+  ##  [username[:password]@][protocol[(address)]]/]
+  ##  see https://github.com/go-sql-driver/mysql#dsn-data-source-name
+  ##  e.g.
+  ##    servers = ["user:passwd@tcp(127.0.0.1:6032)/"]
+  ##    servers = ["user@tcp(127.0.0.1:6032)/"]
+  #
+  ## If no servers are specified, then localhost is used as the host.
+  servers = ["tcp(127.0.0.1:6032)/"]
+
+  ## Selects the metric output format.
+  ##
+  ## This option exists to maintain backwards compatibility, if you have
+  ## existing metrics do not set or change this value until you are ready to
+  ## migrate to the new format.
+  ##
+  ## If you do not have existing metrics from this plugin set to the latest
+  ## version.
+  ##
+  ## Telegraf >=1.6: metric_version = 2
+  ##           <1.6: metric_version = 1 (or unset)
+  metric_version = 2
+
+  ## the limits for metrics form perf_events_statements
+  perf_events_statements_digest_text_limit  = 120
+  perf_events_statements_limit              = 250
+  perf_events_statements_time_limit         = 86400
+  #
+  ## gather metrics from stats.stats_mysql_connection_pool
+  gather_connection_pool                    = true
+  #
+  ## gather metrics from stats.stats_mysql_commands_counters
+  gather_commands_counters                  = true
+  #
+  ## gather metrics from stats.stats_mysql_global
+  gather_global_stats                       = true
+`
+
+func (p *Proxysql) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *Proxysql) Description() string {
+	return "Read metrics from one or many proxysql servers"
+}
+
+func (p *Proxysql) Gather(acc telegraf.Accumulator) error {
+	if len(p.Servers) == 0 {
+		// default to localhost if nothing specified.
+		return p.gatherServer(localhost, acc)
+	}
+
+	var wg sync.WaitGroup
+
+	// Loop through each server and collect metrics
+	for _, server := range p.Servers {
+		wg.Add(1)
+		go func(s string) {
+			defer wg.Done()
+			acc.AddError(p.gatherServer(s, acc))
+		}(server)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (p *Proxysql) gatherServer(serv string, acc telegraf.Accumulator) error {
+	serv, err := dsnAddTimeout(serv)
+	if err != nil {
+		return err
+	}
+
+	db, err := sql.Open("mysql", serv)
+	if err != nil {
+		return err
+	}
+
+	defer db.Close()
+
+	if p.GatherGlobalStats {
+		err = p.gatherGlobalStats(db, serv, acc)
+		if err != nil {
+			return err
+		}
+	}
+
+	if p.GatherConnectionPool {
+		err = p.gatherConnectionPool(db, serv, acc)
+		if err != nil {
+			return err
+		}
+	}
+
+	if p.GatherCommandsCounters {
+		err = p.gatherCommandsCounters(db, serv, acc)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *Proxysql) gatherGlobalStats(db *sql.DB, serv string, acc telegraf.Accumulator) error {
+	rows, err := db.Query(globalStats)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	var (
+		variable_name  string
+		variable_value uint64
+	)
+
+	servtag := getDSNTag(serv)
+
+	for rows.Next() {
+		err = rows.Scan(
+			&variable_name,
+			&variable_value,
+		)
```

Comment:
How about shorten the variable names and write that to one line, e.g.
```suggestion
		err = rows.Scan(&name,	&value)
```